internal/k8shandler/reconciler.go

```diff
@@ -25,6 +30,49 @@ func (er *ElasticsearchRequest) L() logr.Logger {
 	return er.ll
 }
 
+func SecretReconcile(requestCluster *elasticsearchv1.Elasticsearch, requestClient client.Client) error {
+	var secretChanged bool
+
+	nretries := -1
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		nretries++
+		newSecretHash := getSecretDataHash(requestCluster.Name, requestCluster.Namespace, requestClient)
+
+		cluster := &elasticsearchv1.Elasticsearch{}
+		if err := requestClient.Get(context.TODO(), types.NamespacedName{Name: requestCluster.Name, Namespace: requestCluster.Namespace}, cluster); err != nil {
+			return err
+		}
+
+		// compare the new secret with current one in the nodes
+		for _, node := range nodes[nodeMapKey(requestCluster.Name, requestCluster.Namespace)] {
+			if newSecretHash != node.getSecretHash() {
```

Comment:
if the node's `getSecretHash` is empty, this would cause a restart which we don't want...
it could be empty if we've restarted the operator or if the cluster is brand new (neither of those should trigger this)